src/cmd/cli/command/compose.go

```diff
@@ -153,13 +154,18 @@ func makeComposeUpCmd() *cobra.Command {
 			tailOptions := newTailOptionsForDeploy(deploy.Etag, since, verbose)
 			serviceStates, err := cli.TailAndMonitor(ctx, project, provider, time.Duration(waitTimeout)*time.Second, tailOptions)
 			if err != nil {
-				handleTailAndMonitorErr(ctx, err, client, cli.DebugConfig{
+				logs := []string{}
+				if tailOptions.LogCache != nil {
+					logs = tailOptions.LogCache.Get()
+				}
+				handleTailAndMonitorErr(ctx, err, &logs, client, cli.DebugConfig{
```

Comment:
Let's just pass this array directly. I don't think we gain anything by passing a pointer.

```suggestion
				handleTailAndMonitorErr(ctx, err, logs, client, cli.DebugConfig{
```